orquesta/conducting.py

```diff
@@ -882,3 +885,36 @@ def get_task_transition_contexts(self, task_id, route):
                 contexts[task_transition_id] = self.get_task_initial_context(t[1], route)
 
         return contexts
+
+    def request_workflow_rerun(self, rerun_options):
+        # Check current workflow status.
+        if self.get_workflow_status() not in statuses.ABENDED_STATUSES:
+            LOG.debug('Workflow execution cannot rerun from task(s) because it is not in a '
+                      'failed state')
+            return
+
+        # Force reset workflow status to running
+        self.workflow_state.status = statuses.RUNNING
+
+        # Create a new entry in staging for the next task.
+        tasks = self.workflow_state.get_tasks()
+        for i in range(len(tasks)):
+            if tasks[i]['status'] == statuses.FAILED and tasks[i]['id'] in rerun_options:
```

Comment:
Change `for` to the following.  You can't iterate thru tasks because we want to get the most recent task entry for the task_id.

```
for task_id in options.get('tasks', []):
    task_route = 0  # Assume route 0 for now, we need to come back to this for the "split" use case.
    task = self.workflow_state.get_task(task_id, task_route)
    if task['status'] in statuses.ABENDED_STATUSES:
```

Reply:
> Change `for` to the following. You can't iterate thru tasks because we want to get the most recent task entry for the task_id.
 
Changed for loop as suggested here to `for task_id in options.get('tasks', []):`, during unit test, found that if a task name passed from user is invalid,  when `self.workflow_state.get_task(task_id, task_route)` is called, it causes execution
```
Traceback (most recent call last):
  File "/home/vagrant/st2/virtualenv/lib/python2.7/site-packages/orquesta/tests/unit/conducting/test.py", line 121, in test_rerun_simple_workflow_bad_option
    conductor.request_workflow_rerun(options)
  File "/home/vagrant/st2/virtualenv/lib/python2.7/site-packages/orquesta/conducting.py", line 923, in request_workflow_rerun
    task = self.workflow_state.get_task(task_id, task_route)
  File "/home/vagrant/st2/virtualenv/lib/python2.7/site-packages/orquesta/conducting.py", line 71, in get_task
    self.tasks[constants.TASK_STATE_ROUTE_FORMAT % (task_id, str(task_route))]
KeyError: 'task__r0'
```

What can do is make sure all passed task names are valid  by  checking with `tasks = self.workflow_state.get_tasks()` before above `for` loop. Any suggestions?